src/explore-education-statistics-admin/src/pages/release/data/components/ApiDataSetLocationMapModal.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  NewLocationCandidate,
+  LocationMappingWithKey,
+  LocationCandidateWithKey,
+} from '@admin/pages/release/data/utils/getApiDataSetLocationMappings';
+import { PendingLocationMappingUpdate } from '@admin/pages/release/data/ReleaseApiDataSetLocationsMappingPage';
+import ApiDataSetLocationMapForm from '@admin/pages/release/data/components/ApiDataSetLocationMapForm';
+import styles from '@admin/pages/release/data/components/ApiDataSetLocationMapModal.module.scss';
+import ButtonText from '@common/components/ButtonText';
+import Modal from '@common/components/Modal';
+import SummaryList from '@common/components/SummaryList';
+import SummaryListItem from '@common/components/SummaryListItem';
+import VisuallyHidden from '@common/components/VisuallyHidden';
+import useToggle from '@common/hooks/useToggle';
+import { LocationLevelsType } from '@common/utils/locationLevelsMap';
+import React from 'react';
+
+interface Props {
+  candidate?: LocationCandidateWithKey;
+  level: LocationLevelsType;
+  mapping: LocationMappingWithKey;
+  newLocations: NewLocationCandidate[];
+  onSubmit: (update: PendingLocationMappingUpdate) => Promise<void>;
+}
+
+export default function ApiDataSetLocationMapModal({
+  candidate,
+  level,
+  mapping,
+  newLocations = [],
+  onSubmit,
+}: Props) {
+  const [isOpen, toggleOpen] = useToggle(false);
```

Comment:
Was just wondering something about semantics - `useToggle` and `useEffect` seem quite similar. Was just wondering why one is preferred over the other.

Reply:
useToggle is just a simpler way of handling boolean toggles.